src/transformers/models/llama/modeling_llama.py

```diff
@@ -241,7 +241,8 @@ def __init__(self, config: LlamaConfig):
         super().__init__()
         self.config = config
         # For PyTorch/XLA's SPMD 2D sharding
-        self.spmd_2d_sharding = config.spmd_2d_sharding
+        assert config.spmd_2d_sharding & config.spmd_tensor_sharding == 0
```

Comment:
`&` is bitwise `and`, right? So spmd_2d_sharding=2 and spmd_tensor_sharding=4 would pass this test

Reply:
Okay, I guess python land doesn't have the C++ &&...

Reply:
haha, I made the same mistake before as well